### PR TITLE
[STORY-642] new api endpoint to handle password reset

### PR DESCRIPTION
**Pull request by sc-zenokerr**
- [ ] Add a [changelog entry](https://changelog.scalingo.com/)

Added DatabaseUserPasswordReset method

### CODE REVIEW

File: databases.go [hunk 1]
```diff
@@ -319,3 +319,19 @@ func (c *Client) DatabaseListUsers(ctx context.Context, app, addonID string) ([]
 func (c *Client) DatabaseDeleteUser(ctx context.Context, app, addonID, userName string) error {
 	return c.DBAPI(app, addonID).SubresourceDelete(ctx, "databases", addonID, "users", userName)
 }
+
+// DatabaseUserPasswordReset resets the password for a user for given database addon
+func (c *Client) DatabaseUserPasswordReset(ctx context.Context, app, addonID, username string) (DatabaseUser, error) {
```
Comment (EtienneM) [inline databases.go:324]: nitpick(non-blocking): why is the method named `...PasswordReset` whereas the endpoint is `reset_password`? I would have expected the method to be named `...ResetPassword`.

File: databases.go [hunk 2]
```diff
@@ -319,3 +319,19 @@ func (c *Client) DatabaseListUsers(ctx context.Context, app, addonID string) ([]
...
+	res := DatabaseUserResponse{}
+	req := &httpclient.APIRequest{
+		Method:   "PATCH",
+		Endpoint: "/databases/" + addonID + "/users/" + username + "/reset_password",
+		Expected: httpclient.Statuses{200},
```
Comment (EtienneM) [inline databases.go:329]: nitpick(non-blocking): I think in the most recent codes, we eventually prefer to use the constant
```suggestion
		Expected: httpclient.Statuses{http.StatusOK},
```

**Review by EtienneM — APPROVED** [review]
Just a couple of picky suggestions. Feel free to take them into account or not. I'm already approving the PR.

File: databases.go (hunk 1 above)
Comment (sc-zenokerr) [reply databases.go:324]: Indeed... My automatism is to name hierarchically.
Database->User->Password->Reset
but here the action is "reset password".. I'll rename

**Pull request merged**